### PR TITLE
Feature/tri 437

### PR DESCRIPTION
**Pull request by SaberIstos**

### CODE REVIEW

File: irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java [hunk 1]
```diff
@@ -1,8 +1,12 @@
 package net.catenax.irs.connector.job;
 
+import java.util.List;
```
Comment (ds-jhartmann) [inline irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java:3]: unused import

File: irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 
 import com.github.javafaker.Faker;
+import net.catenax.irs.component.enums.AspectType;
```
Comment (ds-jhartmann) [inline irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java:8]: unused import

File: irs-api/src/test/java/net/catenax/irs/util/TestMother.java [hunk 3]
```diff
@@ -51,6 +75,25 @@ public AssemblyPartRelationshipDTO assemblyPartRelationshipDTO() {
                                           .build();
     }
 
+    public JobDataDTO jobDataDTO() {
```
Comment (ds-jhartmann) [inline irs-api/src/test/java/net/catenax/irs/util/TestMother.java:78]: you can make this static like the other methods in this class

File: irs-api/src/test/java/net/catenax/irs/util/TestMother.java [hunk 4]
```diff
@@ -51,6 +75,25 @@ public AssemblyPartRelationshipDTO assemblyPartRelationshipDTO() {
...
+        return JobDataDTO.builder()
+                         .rootItemId("urn:uuid:8a61c8db-561e-4db0-84ec-a693fc5ffdf6")
+                         .treeDepthId("")
+                         .bomLifecycle("AsBuilt")
+                         .aspectTypes(List.of(AspectType.SERIAL_PART_TYPIZATION.toString().toLowerCase(Locale.ROOT),
+                                 AspectType.ASSEMBLY_PART_RELATIONSHIP.toString().toLowerCase(Locale.ROOT)))
+                         .build();
+    }
+
+    public JobDataDTO jobDataFilter() {
```
Comment (ds-jhartmann) [inline irs-api/src/test/java/net/catenax/irs/util/TestMother.java:88]: you can make this static like the other methods in this class

File: irs-parent/pom.xml [hunk 5]
```diff
@@ -34,6 +34,7 @@
         <assertj-core.version>3.21.0</assertj-core.version>
         <gatling.version>3.7.6</gatling.version>
         <resilience4j.version>1.7.1</resilience4j.version>
+        <org.apache.commons.version>4.4</org.apache.commons.version>
```
Comment (ds-jhartmann) [inline irs-parent/pom.xml:37]: You really shouldn't add a new dependency just for a notEmpty function.

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java [hunk 6]
```diff
@@ -62,12 +63,12 @@ private AssemblyPartRelationshipDTO buildAssemblyPartRelationshipResponse(final
...
 
     private void filterSubmodelPartsByLifecycleContext(final Set<ChildData> submodelParts,
-            final String lifecycleContext) {
+                                                       final String lifecycleContext) {
         submodelParts.removeIf(isNotLifecycleContext(lifecycleContext));
     }
 
-    private boolean shouldFilterByLifecycleContext(final String lifecycleContext) {
-        return StringUtils.isNotBlank(lifecycleContext);
+    private boolean shouldFilterByLifecycleContext(final String lifecyleContext) {
```
Comment (ds-jhartmann) [inline irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java:70]: typo: lifecycleContext

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java [hunk 7]
```diff
@@ -62,12 +63,12 @@ private AssemblyPartRelationshipDTO buildAssemblyPartRelationshipResponse(final
...
+        return StringUtils.isNotBlank(lifecyleContext);
```
Comment (ds-jhartmann) [inline irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java:71]: typo: lifecycleContext

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/registry/domain/DigitalTwinRegistryFacade.java [hunk 8]
```diff
@@ -30,32 +34,66 @@ public class DigitalTwinRegistryFacade {
...
+
+        if (shouldFilterByAspectType(aspectTypes)) {
+            final String type = submodelDescriptor.getIdShort();
+            return aspectTypes.contains(type.toLowerCase(Locale.ROOT));
+        }
+        return true;
+    }
+
+    private boolean shouldFilterByAspectType(final List<String> aspectTypes) {
+        return isNotEmpty(aspectTypes);
```
Comment (ds-jhartmann) [inline irs-api/src/main/java/net/catenax/irs/aaswrapper/registry/domain/DigitalTwinRegistryFacade.java:73]: You can replace this imported method with your own.

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
Do you use the right code formatter? The formatter was applied to the whole main branch last friday, so formatting should not change. Please configure IntelliJ to use the formatter stored in this project and push the formatted code.

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASRecursiveJobHandler.java [hunk 9]
```diff
@@ -33,15 +30,15 @@ public AASRecursiveJobHandler(final TreeRecursiveLogic logic) {
...
+        log.info("Initiating request for job {}", job.getJob().getJobId().toString());
+        final var partId = job.getJobData().getRootItemId();
         final var dataRequest = ItemDataRequest.rootNode(partId);
         return Stream.of(dataRequest);
     }
 
     @Override
     public Stream<ItemDataRequest> recurse(final MultiTransferJob job, final AASTransferProcess transferProcess) {
-        log.info("Starting recursive request for job {}", job.getJobIdString());
+        log.info("Starting recursive request for job {}", job.getJob().getJobId().toString());
```
[the remark this answers is not in the document]
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASRecursiveJobHandler.java:41]: log.info("Initiating request for job {}", job.getJobIdString());

File: irs-api/src/main/java/net/catenax/irs/dto/JobDataDTO.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import lombok.Builder;
+import lombok.Data;
+
+/**
+ * JobDataDTO model used for job creation
+ */
+@Data
+@Builder(toBuilder = true)
+@JsonDeserialize(builder = JobDataDTO.JobDataDTOBuilder.class)
+public class JobDataDTO {
```
[the remark this answers is not in the document]
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/dto/JobDataDTO.java:26]: /**
 * JobParameter model used for job creation
 */
@Data
@Builder(toBuilder = true)
@JsonDeserialize(builder = JobParameter.JobParameterBuilder.class)
public class JobParameter {




    /**
     * Builder class
     */
    @JsonPOJOBuilder(withPrefix = "")
    public static class JobParameterBuilder {
    }

File: irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java [hunk 11]
```diff
@@ -60,8 +60,8 @@ public class MultiTransferJob {
      * Arbitrary data attached to the job.
      */
     @Getter
-    @Singular("jobDatum")
-    private Map<String, String> jobData;
+/*    @Singular("jobDatum")*/
```
[the remark this answers is not in the document]
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java:63]: /**
     * Arbitrary data attached to the job.
     */
    @Getter
    private JobParameter jobData;

File: irs-api/src/test/java/net/catenax/irs/aaswrapper/job/AASTransferProcessManagerTest.java [hunk 12]
```diff
@@ -15,12 +15,15 @@
...
 import net.catenax.irs.connector.job.ResponseStatus;
 import net.catenax.irs.connector.job.TransferInitiateResponse;
+import net.catenax.irs.util.TestMother;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(MockitoExtension.class)
-class AASTransferProcessManagerTest {
+public class AASTransferProcessManagerTest {
```
[the remark this answers is not in the document]
Comment (SaberIstos) [reply irs-api/src/test/java/net/catenax/irs/aaswrapper/job/AASTransferProcessManagerTest.java:24]: @ExtendWith(MockitoExtension.class)
class AASTransferProcessManagerTest {

File: irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java (hunk 2 above)
Comment (SaberIstos) [reply irs-api/src/test/java/net/catenax/irs/connector/job/MultiTransferJobTest.java:8]: package net.catenax.irs.connector.job;

import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

import com.github.javafaker.Faker;
import net.catenax.irs.dto.JobParameter;
import net.catenax.irs.util.TestMother;
import org.junit.jupiter.api.Test;

class MultiTransferJobTest {

File: irs-api/src/test/java/net/catenax/irs/util/TestMother.java (hunk 4 above)
Comment (SaberIstos) [reply irs-api/src/test/java/net/catenax/irs/util/TestMother.java:88]: public static JobParameter jobParameter() {
        return JobParameter.builder()
                           .rootItemId("urn:uuid:8a61c8db-561e-4db0-84ec-a693fc5ffdf6")
                           .treeDepthId("")
                           .bomLifecycle("AsBuilt")
                           .aspectTypes(List.of(AspectType.SERIAL_PART_TYPIZATION.toString().toLowerCase(Locale.ROOT),
                                 AspectType.ASSEMBLY_PART_RELATIONSHIP.toString().toLowerCase(Locale.ROOT)))
                           .build();
    }

    public static JobParameter jobParameterFilter() {
        return JobParameter.builder()
                           .rootItemId("urn:uuid:8a61c8db-561e-4db0-84ec-a693fc5ffdf6")
                           .treeDepthId("")
                           .bomLifecycle("AsRequired")
                           .aspectTypes(List.of(AspectType.MATERIAL_ASPECT.toString().toLowerCase(Locale.ROOT)))
                           .build();
    }

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java (hunk 6 above)
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java:70]: private boolean shouldFilterByLifecycleContext(final String lifecycleContext) {
        return StringUtils.isNotBlank(lifecycleContext);
    }

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java (hunk 7 above)
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/aaswrapper/submodel/domain/SubmodelFacade.java:71]: private boolean shouldFilterByLifecycleContext(final String lifecycleContext) {
        return StringUtils.isNotBlank(lifecycleContext);
    }

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/registry/domain/DigitalTwinRegistryFacade.java (hunk 8 above)
Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/aaswrapper/registry/domain/DigitalTwinRegistryFacade.java:73]: private boolean shouldFilterByAspectType(final List<String> aspectTypes) {
        return aspectTypes != null && aspectTypes.size() > 0;
    }

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]

File: .idea/codeStyles/codeStyleConfig.xml [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <state>
-    <option name="USE_PER_PROJECT_SETTINGS" value="true" />
```
[the remark this answers is not in the document]
Comment (SaberIstos) [reply .idea/codeStyles/codeStyleConfig.xml:3]: Updated with the main branche state:
<component name="ProjectCodeStyleConfiguration">
  <state>
    <option name="USE_PER_PROJECT_SETTINGS" value="true" />
  </state>
</component>

File: irs-api/src/main/java/net/catenax/irs/dto/JobParameter.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Job Data key for root item ID
+     */
+    private String rootItemId;
+
+    /**
+     * Expected depth of the tree
+     */
+    private String treeDepth;
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/dto/JobParameter.java:36]: Also, why is this a string? :D

Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/dto/JobParameter.java:36]: In the previous main code, we're using a Map<String, String> where we convert to String:
paramMap.put(DEPTH_ID_KEY, String.valueOf(request.getDepth()));

I didn't change to Integer.

Do you see any side effect setting directly to private Integer treeDepth; ?

Comment (ds-jkreutzfeld) [reply irs-api/src/main/java/net/catenax/irs/dto/JobParameter.java:36]: The map was pretty generic, so String was just the common denominator.

As we defined the depth to always be > 0 and <= 100, I think "int" should be fine (don't need null here with "Integer").

Comment (SaberIstos) [reply irs-api/src/main/java/net/catenax/irs/dto/JobParameter.java]: /**
     * Expected depth of the tree
     */
    private int treeDepth;

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASRecursiveJobHandler.java [hunk 15]
```diff
@@ -63,8 +60,8 @@ public void complete(final MultiTransferJob job) {
         logic.assemblePartialItemGraphBlobs(completedTransfers, targetBlobName.toString());
     }
 
-    private Integer getExpectedTreeDepth(final Map<String, String> jobData) {
-        return Integer.parseInt(jobData.get(DEPTH_ID_KEY));
+    private int getExpectedTreeDepth(final JobParameter jobData) {
+        return jobData.getTreeDepth();
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASRecursiveJobHandler.java:64]: Don't know if this method is really necessary now anymore.

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**